File: src/main/java/net/Chidoziealways/everythingjapanese/entity/client/ModModelLayers.java
package net.Chidoziealways.everythingjapanese.entity.client;

import com.google.common.collect.Sets;
import net.Chidoziealways.everythingjapanese.EverythingJapanese;
import net.minecraft.client.model.geom.ModelLayerLocation;
import net.minecraft.resources.ResourceLocation;

import java.util.Set;

public class ModModelLayers {
    private static final Set<ModelLayerLocation> ALL_MODELS = Sets.newHashSet();
    public static final ModelLayerLocation SIKA_DEER = register("sika_deer");
    public static final ModelLayerLocation SIKA_DEER_BABY = register("sika_deer_baby");
    public static final ModelLayerLocation TRICERATOPS = register("triceratops");
    public static final ModelLayerLocation TRICERATOPS_BABY = register("triceratops_baby");


    private static ModelLayerLocation register(String pPath) {
        return register(pPath, "main");
    }

    private static ModelLayerLocation register(String pPath, String pModel) {
        ModelLayerLocation modellayerlocation = createLocation(pPath, pModel);
        if (!ALL_MODELS.add(modellayerlocation)) {
            throw new IllegalStateException("Duplicate registration for " + modellayerlocation);
        } else {
            return modellayerlocation;
        }
    }

    private static ModelLayerLocation createLocation(String pPath, String pModel) {
        return new ModelLayerLocation(ResourceLocation.fromNamespaceAndPath(EverythingJapanese.MOD_ID, pPath), pModel);
    }
}

File: src/main/kotlin/net/Chidoziealways/everythingjapanese/entity/client/ModModelLayers.kt
package net.Chidoziealways.everythingjapanese.entity.client

import com.google.common.collect.Sets
import net.Chidoziealways.everythingjapanese.EverythingJapanese
import net.Chidoziealways.everythingjapanese.MOD_ID
import net.minecraft.client.model.geom.ModelLayerLocation
import net.minecraft.resources.ResourceLocation

object ModModelLayers {
    private val ALL_MODELS: MutableSet<ModelLayerLocation?> = Sets.newHashSet<ModelLayerLocation?>()
    val SIKA_DEER: ModelLayerLocation = register("sika_deer")
    val SIKA_DEER_BABY: ModelLayerLocation = register("sika_deer_baby")
    val TRICERATOPS: ModelLayerLocation = register("triceratops")
    val TRICERATOPS_BABY: ModelLayerLocation = register("triceratops_baby")


    private fun register(pPath: String, pModel: String = "main"): ModelLayerLocation {
        val modellayerlocation = createLocation(pPath, pModel)
        check(ALL_MODELS.add(modellayerlocation)) { "Duplicate registration for " + modellayerlocation }
        return modellayerlocation
    }

    private fun createLocation(pPath: String, pModel: String): ModelLayerLocation {
        return ModelLayerLocation(ResourceLocation.fromNamespaceAndPath(MOD_ID, pPath), pModel)
    }
}
